src/trimfastq.py

```diff
@@ -0,0 +1,247 @@
+##################################
+#                                #
+# Last modified 2017/11/08       # 
+#                                #
+# Georgi Marinov                 #
+#                                # 
+##################################
+
+import sys
+import os
+
+# try:
+# 	import psyco
+# 	psyco.full()
+# except:
+# 	pass
+
+def run():
+
+    if len(sys.argv) < 2:
+        print('usage: python %s <inputfilename> <bpToKeep | max> [-trim5 bp] [-flowcellID flowcell] [-addEnd 1 | 2] [-replace string newstring | blank] [-renameIDs prefix] [-stdout]' % sys.argv[0])
+        print('\tthe -trim5 option will trim additional bp from the 5 end, i.e. if you want the middle 36bp of 38bp reads, use 36 as bp to keep and 1 as the trim5 argument')
+        print('\tUse - to specify standard input, the script will print(to standard output by default')
+        print('\tThe script can read compressed files as long as they have the correct suffix - .bz2 or .gz')
+        sys.exit(1)
+
+    inputfilename = sys.argv[1]
+    doMax=False
+    if sys.argv[2] == 'max':
+        doMax=True
+        trim='max'
+    else: 
+        trim = int(sys.argv[2])
+    outputfilename = inputfilename.split('/')[-1].split('.fastq')[0] + '.' +str(trim)+'mers.fastq'
+    doFlowcellID=False
+
+    doStdOut=True
+#    doStdOut=False
+#    if '-stdout' in sys.argv:
+#        doStdOut = True
+
+    if '-flowcellID' in sys.argv:
+        doFlowcellID=True
+        flowcellID=sys.argv[sys.argv.index('-flowcellID')+1]
+        if doStdOut:
+            pass
+        else:
+            print('will include flowcell ID', flowcellID, 'in reads headers')
+
+    doRenameIDs = False
+    if '-renameIDs' in sys.argv:
+        doRenameIDs = True
+        RID = '@' + sys.argv[sys.argv.index('-renameIDs') + 1]
+
+    dotrim5=False
+    if '-trim5' in sys.argv:
+        dotrim5=True
+        trim5=int(sys.argv[sys.argv.index('-trim5')+1])
+        if doStdOut:
+            pass
+        else:
+            print('will trim ', trim5, 'bp from the 5-end')
+        outputfilename = inputfilename.split('.fastq')[0] + '.' +str(trim)+'bp-5prim-trim.fastq'
+
+    doAddEnd=False
+    if '-addEnd' in sys.argv:
+        doAddEnd=True
+        END=sys.argv[sys.argv.index('-addEnd')+1]
+        if doStdOut:
+            pass
+        else:
+            print('will add',  '/'+END, 'to read IDs')
+
+    doReplace=False
+    if '-replace' in sys.argv:
+        doReplace=True
+        oldstring=sys.argv[sys.argv.index('-replace')+1]
+        newstring=sys.argv[sys.argv.index('-replace')+2]
+        if newstring == 'blank':
+            newstring=''
+        if doStdOut:
+            pass
+        else:
+            print('will replace',  oldstring, 'with', newstring, 'in read IDs')
+
+    i=0 
```

Comment:
Move definition of `i` and `shorter` closer to where they are used.

Reply:
`i` and `shorter` are used everywhere in the code. I moved it like 20 lines but couldn't move it further.